client/src/Components/Layouts/Captains/EditCaptain/editCaptain.jsx

```diff
@@ -0,0 +1,301 @@
+            message: 'فشل',
+            description: 'هناك خطأ ما الرجاء اعادة ارسال البيانات',
+            icon: <Icon type="meh" style={{ color: '#108ee9' }} />,
+          })
+        }
+        );
+
+
+    }
+    render() {
+      const uploadButton = (
+        <Button className="btn--upload">
+
+          <Icon type={this.state.loading ? 'check-circle' : 'upload'} />
+          أرفق صورة
+        </Button>
+
+      );
+      const imageUrl = this.state.imageUrl;
+      const photoUrl = this.state.photoUrl;
+      const { visible, onCancel, onCreate, form } = this.props;
+      const { getFieldDecorator } = form;
+      return (
+        <Modal
+          visible={visible}
+          title="تعديل كابتن"
+          okText="حفظ"
+          onCancel={onCancel}
+          onOk={onCreate}
+          cancelText="إلغاء"
+          className="edit-captain__popup"
+        >
+          <Form className='edit-captain__form'>
+            <div className="edit-captain-container-left">
+              <Form.Item label="رقم الهوية" >
+                {getFieldDecorator("IDNumber", {
+                  rules: [
+                    { required: true, message: "يرجى ملئ الحقل بارقام ", pattern: /^[0-9]{9}$/ }
+                  ]
+                })(<Input type="text" id="IDNumber" />)}
+              </Form.Item>
+              <Form.Item label="رقم الرخصة">
+                {getFieldDecorator("licenceNumber", {
+                  rules: [
+                    { required: true, message: "يرجى ملئ الحقل بارقام ", pattern: /^[0-9]{7}$/ }
+                  ]
+                })(<Input type="text" id="licenceNumber" />)}
+              </Form.Item>
+              <Form.Item label="الحالة">
+                {getFieldDecorator("status", {
+                  rules: [
+                    { required: true, message: "يرجى اختيار حالة" }
+                  ]
+                })(
+                  <Select style={{ width: 80 }} id='status'>
+                    <Option value="true">فعال </Option>
+                    <Option value="false">غير فعال</Option>
+                  </Select>
+                )}
+              </Form.Item>
+              <Form.Item label="صورة الهوية">
+                {getFieldDecorator("file", {
+                  rules: [
+                    {
+                      required: true,
+                      message: "يرجى رفع صورة الكابتن"
+                    }
+                  ]
+                })(<Upload
+                  accept=".jpg , .png, .jpeg"
+                  name="file"
+                  className="avatar-uploader"
+                  showUploadList={false}
+                  customRequest={this.handleReuest}
+                  multiple={false}
+                >
+                  {uploadButton}
+                  <div className="image__name">
+                    {(imageUrl) ? imageUrl && <><Icon type="check-circle" />{imageUrl.name}</> : photoUrl && <><Icon /><img src={photoUrl} className="upload-photo" /></>}
```

Comment:
@naremanhilles To make this more readable, you could create a few helper functions, and then change the line to call them:
`{ imageUrl ? this.renderImageUrl() : this.renderPhotoUrl() }`
and then somewhere else in your class you can add the helper functions:
`renderImageUrl = () => <><Icon type="check-circle" />{this.state.imageUrl.name}</>`
`renderPhotoUrl = () => this.state.photoUrl && <><Icon /><img src={this.state.photoUrl} className="upload-photo" /></>`

Also you can see that I removed the `imageUrl &&` because the first part of that line (the `imageUrl ?` is already ensuring that imageUrl is truthy.